datafusion/core/src/physical_planner.rs

```diff
@@ -1014,14 +1014,21 @@ impl DefaultPhysicalPlanner {
                             })
                             .collect();
 
+                        let metadata: HashMap<_, _> = left_df_schema
```

Comment:
this one we should be able to write a test for -- perhaps with a query with a JOIN in https://github.com/apache/datafusion/blob/main/datafusion/sqllogictest/test_files/metadata.slt